onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -179,11 +179,17 @@ Status ReadExternalDataForTensor(const ONNX_NAMESPACE::TensorProto& tensor_proto
       GetExternalDataInfo(tensor_proto, tensor_proto_dir, external_file_path, file_offset, tensor_byte_size));
 
   unpacked_tensor.resize(tensor_byte_size);
-  ORT_RETURN_IF_ERROR(onnxruntime::Env::Default().ReadFileIntoBuffer(
-      external_file_path.c_str(),
-      file_offset,
-      tensor_byte_size,
-      gsl::make_span(reinterpret_cast<char*>(unpacked_tensor.data()), tensor_byte_size)));
+  if (external_file_path == onnxruntime::utils::kTensorProtoMemoryAddressTag) {
+    // the value in location is the memory address of the data
+    const void* ext_data_buf = reinterpret_cast<const void*>(file_offset);
+    std::memcpy(unpacked_tensor.data(), ext_data_buf, tensor_byte_size);
+  } else {
+    ORT_RETURN_IF_ERROR(onnxruntime::Env::Default().ReadFileIntoBuffer(
+        external_file_path.c_str(),
+        file_offset,
+        tensor_byte_size,
+        gsl::make_span(reinterpret_cast<char*>(unpacked_tensor.data()), tensor_byte_size)));
+  }
```

Comment:
Note: fixes incomplete handling of external data that is stored in memory with the tag `kTensorProtoMemoryAddressTag`. I ran into this when serializing an OrtModel (created via editor api) to a model proto. There's a test in this PR that triggers this scenario.

Reply:
This is re-worked in my OrtValue initializers PR